niceman/resource/ssh.py

```diff
@@ -46,15 +56,16 @@ class SSH(Resource):
     _ssh = attr.ib(default=None)
 
     def connect(self):
+        """Open a connection to the environment resource.
         """
-        Open a connection to the environment resource.
-        """
-        self._ssh = SSHClient(
+        self._ssh = paramiko.SSHClient()
+        self._ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
```

Comment:
TODO (file an issue): we shouldn't blindly login to unknown hosts

Reply:
didn't look how difficult or possible via paramiko but we need make sure that it works coherently with existing system/user ssh configuration, i.e. that existing known_hosts used by user's `ssh` command are used, which seems to be not the case in simplest paramiko invocation:

```
$> python -c 'from paramiko import SSHClient; ssh = SSHClient().connect("127.0.0.1", username=None)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/paramiko/client.py", line 361, in connect
    server_key)
  File "/usr/lib/python2.7/dist-packages/paramiko/client.py", line 648, in missing_host_key
    raise SSHException('Server %r not found in known_hosts' % hostname)
paramiko.ssh_exception.SSHException: Server '127.0.0.1' not found in known_hosts
```
- make sure that ssh keys authentication (possibly provided by ssh-agent) would be used